pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -132,34 +155,35 @@ func (c *Cluster) Start() error {
 	}()
 
 	if c.Storage.UseAllNodes {
```

Comment:
`Start()` is a huge function with everything added here. Could it be factored out into multiple functions for readability?

Reply:
+1, there are lots of room for readability here. It is a bit fragile to do it here though. I added a followup item.